presto-main/src/test/java/io/prestosql/server/TestQueryResource.java

```diff
@@ -89,6 +106,17 @@ public void testGetQueryInfos()
         infos = getQueryInfos("/v1/query?state=running");
         assertEquals(infos.size(), 0);
         assertStateCounts(infos, 0, 0, 0);
+
+        server.getAccessControl().deny(privilege("query", VIEW_QUERY));
+        try {
+            assertTrue(getQueryInfos("/v1/query").isEmpty());
+            assertTrue(getQueryInfos("/v1/query?state=finished").isEmpty());
+            assertTrue(getQueryInfos("/v1/query?state=failed").isEmpty());
+            assertTrue(getQueryInfos("/v1/query?state=running").isEmpty());
+        }
+        finally {
+            server.getAccessControl().reset();
```

Comment:
It would be nice to have some closeable entity.

```
try (AutoCloseasble privilege = deny(privilege("query", VIEW_QUERY)) { 
   ...
}

...
private AutoCloseable deny(... privilege) {
   server.getAccessControl().deny(privilege);
   return () -> server.getAccessControl().reset();
}
```